models/migrations/v213.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"fmt"
+
+	"xorm.io/xorm"
+)
+
+func createRepoSettingsTable(x *xorm.Engine) error {
+	type RepoSetting struct {
+		ID           int64  `xorm:"pk autoincr"`
+		RepoID       int64  `xorm:"index unique(key_repoid)"`              // to load all of someone's settings
+		SettingKey   string `xorm:"varchar(255) index unique(key_repoid)"` // ensure key is always lowercase
+		SettingValue string `xorm:"text"`
```

Comment:
Something I already noticed for the user settings k/v table: Shouldn't this be 
```suggestion
		SettingValue string `xorm:"longtext"`
```
instead, so that it works as intended even on MySQL?

Reply:
Also, if we change the text value here, should we change it for the user settings in this migration as well?

Reply:
Take a look at current setting in `app.ini`, no value are more than 16K so I think `text` is big enough?

Reply:
Because at the moment, no objects are serialized into the table.
However, it is not impossible that objects should be serialized there at some point (see i.e. #19831 or #2175).
In that case, we might suddenly get an unintended bug.